pkg/shed/db.go

```diff
@@ -23,45 +23,58 @@
 package shed
 
 import (
+	"bytes"
+	"context"
+	"errors"
+
+	"github.com/dgraph-io/badger/v2"
 	"github.com/ethersphere/bee/pkg/logging"
-	"github.com/syndtr/goleveldb/leveldb"
-	"github.com/syndtr/goleveldb/leveldb/iterator"
-	"github.com/syndtr/goleveldb/leveldb/opt"
 )
 
 const (
-	openFileLimit = 128 // The limit for LevelDB OpenFilesCacheCapacity.
+	DefaultSyncWrites         = false  // Dont sync the writes to disk, instead delay it as a batch
+	DefaultValueThreshold     = 1024   // Anything less than 1K value will be store with the LSM key itself
+	DefaultValueLogMaxEntries = 500000 // Max number of entries in a value log
+)
+
+var (
+	ErrNotFound = errors.New("storage: not found")
 )
 
-// DB provides abstractions over LevelDB in order to
+// DB provides abstractions over badgerDB in order to
 // implement complex structures using fields and ordered indexes.
 // It provides a schema functionality to store fields and indexes
 // information about naming and types.
 type DB struct {
-	ldb     *leveldb.DB
+	path    string
+	bdb     *badger.DB
 	metrics metrics
 	logger  logging.Logger
-	quit    chan struct{} // Quit channel to stop the metrics collection before closing the database
 }
 
-// NewDB constructs a new DB and validates the schema
-// if it exists in database on the given path.
-// metricsPrefix is used for metrics collection for the given DB.
+// NewDB opens the badger DB with options that make the DB useful for
+// Chunk, State as well as Index stores
 func NewDB(path string, logger logging.Logger) (db *DB, err error) {
-	ldb, err := leveldb.OpenFile(path, &opt.Options{
-		OpenFilesCacheCapacity: openFileLimit,
-	})
+	o := badger.DefaultOptions(path)
+	o.SyncWrites = DefaultSyncWrites
+	o.ValueThreshold = DefaultValueThreshold
+	o.ValueLogMaxEntries = DefaultValueLogMaxEntries
+	o.Logger = nil // Dont enable the badger logs
+	database, err := badger.Open(o)
 	if err != nil {
+		logger.Error("could not open database.")
```

Comment:
not needed to log _and_ return the error. enough just to return the error in this case